ext/distributed_tracing_headers.c

```diff
@@ -42,6 +43,54 @@ static void dd_check_tid(ddtrace_distributed_tracing_result *result) {
     }
 }
 
+static void ddtrace_deserialize_baggage(char *baggage_ptr, char *baggage_end, HashTable *baggage) {
+    while (baggage_ptr < baggage_end) {
+        char *key_start = baggage_ptr;
+        while (baggage_ptr < baggage_end && *baggage_ptr != '=') {
+            ++baggage_ptr;
+        }
+        if (baggage_ptr >= baggage_end || *baggage_ptr != '=') break;
+
+        size_t key_len = baggage_ptr - key_start;
+        ++baggage_ptr; // skip '='
+
+        char *value_start = baggage_ptr;
+        while (baggage_ptr < baggage_end && *baggage_ptr != ',') {
+            ++baggage_ptr;
+        }
+
+        size_t value_len = baggage_ptr - value_start;
+        if (key_len > 0 && value_len > 0) {
+            zend_string *key = zend_string_init(key_start, key_len, 0);
+            zval value;
+            ZVAL_STRINGL(&value, value_start, value_len);
+            zend_hash_update(baggage, key, &value);
+            zend_string_release(key);
+        }
+
+        if (baggage_ptr < baggage_end && *baggage_ptr == ',') {
+            ++baggage_ptr; // skip ','
+        }
```

Comment:
```suggestion
        ++baggage_ptr; // skip ','
```
We can skip this unconditionally (the offset is going to be checked at the start of the loop)